Gems/ROS2/Code/Include/ROS2/Sensor/Events/EventSourceAdapter.h

```diff
@@ -89,57 +86,55 @@ namespace ROS2
                 {
-                    const AZStd::chrono::duration<float, AZStd::chrono::seconds::period> expectedLoopTime =
-                        ROS2Interface::Get()->GetSimulationClock().GetExpectedSimulationLoopTime();
+                    const float sourceDeltaTime = m_eventSource.GetDeltaTime(AZStd::forward<decltype(args)>(args)...);
+                    m_adaptedDeltaTime += sourceDeltaTime;
 
-                    if (IsPublicationDeadline(expectedLoopTime.count()))
+                    if (!IsPublicationDeadline(sourceDeltaTime))
                     {
-                        m_sensorAdaptedEvent.Signal(args...);
+                        return;
                     }
+
+                    m_sensorAdaptedEvent.Signal(m_adaptedDeltaTime, AZStd::forward<decltype(args)>(args)...);
+                    m_adaptedDeltaTime = 0.0f;
                 });
-            m_eventSource.ConnectToSourceEvent(m_adaptedEventHandler);
-            m_eventSource.Activate();
+            m_eventSource.ConnectToSourceEvent(m_sourceAdaptingEventHandler);
+            m_eventSource.Start();
         }
 
-        //! Deactivates event source adapter - deactivates event source and disconnects internal adapted event handler from source event. If
-        //! it will be necessary, this implementation allows multiple consecutive calls of Activate / Deactivate, however user must also
-        //! investigate specific event source implementation with such case in mind.
-        void Deactivate()
+        //! Stops event source adapter - stops event source and disconnects internal adapted event handler from source event. If it will be
+        //! necessary, this implementation allows multiple consecutive calls of Start / Stop, however user must also investigate specific
+        //! event source implementation with such case in mind.
+        void Stop()
         {
-            m_eventSource.Deactivate();
-            m_adaptedEventHandler.Disconnect();
+            m_eventSource.Stop();
+            m_sourceAdaptingEventHandler.Disconnect();
         }
 
-        //! Parses sensor configuration into event source configuration.
-        //! @param sensorConfiguration Configuration of sensor using this event source adapter.
-        void Configure(const SensorConfiguration& sensorConfiguration)
+        //! Sets adapter working frequency. By design, adapter will not work correctly, if this frequency will be greater than used event
+        //! source frequency - e.g. adapter will be requested to work in 60Hz, when using event source working in 30Hz. In general, adapted
+        //! frequency should be equal or lower than event source frequency. Optimal (highest precision in timing events) working conditions
+        //! take place when event source frequency is an integer multiple of adapted frequency.
+        //! @param adaptedFrequency Adapter working frequency. When set to zero or less adapter will be assumed to work in 1Hz.
```

Comment:
This an unnecessarily limiting assumption. Once per day is a sensible minimum (about 0.000012). This needs to be applied in SensorConfiguration m_frequency field MinValue Attribute. Please raise a new issue to limit frequency input (also consider limiting max to what the global setting for physics currently allows).